pkg/kubelet/images/puller.go

```diff
@@ -58,8 +59,14 @@ func (pip *parallelImagePuller) pullImage(ctx context.Context, spec kubecontaine
 		}
 		startTime := time.Now()
 		imageRef, err := pip.imageService.PullImage(ctx, spec, pullSecrets, podSandboxConfig)
+		var size uint64
+		if err == nil && imageRef != "" {
+			// Getting the image size with best effort, ignoring the error.
+			size, _ = pip.imageService.GetImageSize(ctx, spec)
+		}
 		pullChan <- pullResult{
 			imageRef:     imageRef,
+			imageSize:    size,
 			err:          err,
 			pullDuration: time.Since(startTime),
```

Comment:
thinking out loud - since getting the size if not a part of download operation, I wonder if it will be better to calculate time before the GetImageSize call. On other side, now we are spending this time to get size as part of the pull operation and this is what end user will actually see. So perhaps it should be accounted same way as we account for the wait time.

Maybe you can add a comment here noting this for future. If we will decide to calculate pure download time without the wait we may decide to cut the time of GetImageSize from it as well

Reply:
This duration here is more of an e2e latency from kubelet's view. We already included the latency of credential resolution here. So I am leaning towards including getting image size here.

Added a comment.